move-to-infra/move-to-infra.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+
+# This script moves ARO apps into intrastructure nodes.
+# It follows RHEL guide: https://access.redhat.com/solutions/5034771
+#
+# Infra nodes are:
+#  label: node-role.kubernetes.io/infra: ""
+#  taints:
+#  - key: infra
+#    value: reserved
+#    effect: NoSchedule
+#  - key: infra
+#    value: reserved
+#    effect: NoExecute
+
+
+# Check if pods are running
+checkStatus()
+{
+  LABEL="$1"
+  NAMESPACE="$2"
+  EXPECTED_PODS="$3"
+  
+  echo "...checking pod labels ${LABEL} at namespace ${NAMESPACE}..."
+  ATTEMPTS=0
+  until [[ $ATTEMPTS -ge 10 ]]
+  do
+    PODS_READY=$(oc get pod  -l "${1}" --field-selector=status.phase==Running -o jsonpath="{.items[*].metadata.name}" -n${NAMESPACE} | wc -w)
+    if [[ $PODS_READY -eq $EXPECTED_PODS ]]; then
+      echo "...found expected pods running"
+      break
+    fi
+    ATTEMPTS=$[$ATTEMPTS+1]
+    echo "Sleep 10s and check again."
+    sleep 10
+  done
+  if [[ $PODS_READY -ne $EXPECTED_PODS ]]; then
+    echo "...not found expected pods. exit"
+    exit 1
+  fi
+}
+
+# create config map to move 
+monitoringConfigMap()
+{
+  cat <<EOT > cluster-monitoring-configmap.yaml
```

Comment:
Is there any chance we can extract an inline yaml definition into a separate file? That should improve the readability of the code and allow us to run a simple tyaml linting and shellchecks against files with pre-commit

Reply:
You can apply the yaml directly within this function. E.g.:
cat <<EOF | oc apply -f -
apiVersion: v1
...
EOF

Reply:
Having yaml file inline definitely has its own pros.
But the concern I have, is that YAML files can be tested if they're defined inline. But extracting it into separate files you get the following benefits:
1. ability to lint yamls: https://github.com/pre-commit/pre-commit-hooks#check-yaml
2. increase readability of shell files
3. reduced cognitive load when updating/maintaining scripts

Reply:
Ok, moved it to separate file but file is get from RHEL case 1-1.